core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -1058,12 +1058,17 @@ public List<TableIdentifier> listViews(SessionContext context, Namespace namespa
   public View loadView(SessionContext context, TableIdentifier identifier) {
     checkViewIdentifierIsValid(identifier);
 
-    LoadViewResponse response =
-        client.get(
-            paths.view(identifier),
-            LoadViewResponse.class,
-            headers(context),
-            ErrorHandlers.viewErrorHandler());
+    LoadViewResponse response;
+    try {
+      response =
+          client.get(
+              paths.view(identifier),
+              LoadViewResponse.class,
+              headers(context),
+              ErrorHandlers.viewErrorHandler());
+    } catch (UnsupportedOperationException | RESTException e) {
+      throw new NoSuchViewException("Service does not support views");
```

Comment:
I don't feel this message correctly represents the possible reasons why this might fail.  I'd suggest trying to scope this to 4xx (403, 404) and change it to "Unable to load view: <view>".

Reply:
Yeah, I think you're right.